File: src/main/java/io/github/saneea/dvh/feature/multi/FeatureCatalogPrinter.java
package io.github.saneea.dvh.feature.multi;

import java.io.PrintStream;

import io.github.saneea.dvh.FeatureContext;
import io.github.saneea.dvh.FeatureProvider;

public abstract class FeatureCatalogPrinter {
	protected final PrintStream out;

	public FeatureCatalogPrinter(PrintStream out) {
		this.out = out;
	}

	abstract void print(FeatureProvider featureProvider, FeatureContext context);
}

File: src/main/java/io/github/saneea/dvh/feature/multi/FeatureCatalogPrinter.kt
package io.github.saneea.dvh.feature.multi

import io.github.saneea.dvh.FeatureContext
import io.github.saneea.dvh.FeatureProvider
import java.io.PrintStream

abstract class FeatureCatalogPrinter(protected val out: PrintStream) {
    abstract fun print(featureProvider: FeatureProvider, context: FeatureContext)
}